### PR TITLE
[14.0][MIG] website_oca_integrator: Migration to 14.0

### PR DESCRIPTION
**Pull request by ernestotejeda**
Cc @Tecnativa TT30063

Main changes:
- Adapt python code to use *PyGithub* library as it is used in the module *github_connector* (https://github.com/OCA/interface-github/tree/14.0/github_connector).
- Adapt tests to the use of *PyGithub* library.
- Remove demo data as those depends on the demo data that was removed in the *github-connector* migration to v14.0 and without them it not make sense. See https://github.com/OCA/interface-github/pull/73#issuecomment-875572690
- Adapt js code to ES6.

### CODE REVIEW

**Review by CarlosRoca13 — APPROVED** [review]

File: oca_dependencies.txt [hunk 1]
```diff
@@ -1 +1,2 @@
 interface-github
+apps-store https://github.com/Tecnativa/apps-store 14.0-mig-apps_product_creator
```
Comment (victoralmau) [inline oca_dependencies.txt]:
```suggestion
apps-store
```

File: website_oca_integrator/controllers/member.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+from odoo.addons.website_membership.controllers.main import WebsiteMembership
+
+
+class WebsiteContributorMembership(WebsiteMembership):
+    @http.route()
+    def partners_detail(self, partner_id, **post):
+        """
+        Display contributor/member page.
+        """
+        response = super(WebsiteContributorMembership, self).partners_detail(
```
Comment (victoralmau) [inline website_oca_integrator/controllers/member.py]:
```suggestion
        response = super().partners_detail(
```

File: website_oca_integrator/controllers/portal.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        modules_data = [{"id": m.id, "name": m.name} for m in modules]
+        return json.dumps(modules_data)
+
+    def details_form_validate(self, data):
+        # after adding HTML editor in portal page, if we click on
+        # 'Confirm' button then, 'files' key is passed in post data.
+        # this field does not exist in 'res.partner'. removed this
+        # key to prevent an error of "Unknown field 'files'".
+        data.pop("files", False)
+        error, error_message = super(IntegratorPortal, self).details_form_validate(data)
```
Comment (victoralmau) [inline website_oca_integrator/controllers/portal.py]:
```suggestion
        error, error_message = super().details_form_validate(data)
```

File: website_oca_integrator/controllers/website_sale.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        else:
+            ppg = PPG
+        return ppg, post
+
+    @http.route()
+    def shop(self, page=0, category=None, search="", integrator="", ppg=False, **post):
+        """
+        Filter products by integrator.
+        """
+        response = super(WebsiteIntegratorSale, self).shop(
```
Comment (victoralmau) [inline website_oca_integrator/controllers/website_sale.py]:
```suggestion
        response = super().shop(
```

File: website_oca_integrator/tests/test_ui_portal.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+
+@odoo.tests.tagged("post_install", "-at_install")
+class TestUi(odoo.tests.HttpCase):
+    def setUp(self):
+        super().setUp()
+        # Trick this configuration value for avoiding an error
+        config["source_code_local_path"] = "/tmp/"
+        # HACK: Force computation due to demo data glitch
+        # partner = self.env.ref("website_oca_integrator.partner_integrator_portal_demo")
```
Comment (victoralmau) [inline website_oca_integrator/tests/test_ui_portal.py]:
```suggestion
```

**Review by victoralmau — CHANGES_REQUESTED** [review]

**Comment by ernestotejeda** [comment]
@victoralmau changes done

**Review by victoralmau — APPROVED** [review]

**Pull request merged**